lib/shelf_io.dart

```diff
@@ -133,7 +133,8 @@ Request _fromHttpRequest(HttpRequest request) {
       protocolVersion: request.protocolVersion,
       headers: headers,
       body: request,
-      onHijack: onHijack);
+      onHijack: onHijack,
+      context: {'io.httprequest.connectionInfo': request.connectionInfo});
```

Comment:
I think I'd call this field `"shelf.io.connection_info"`. We want to make sure it's scoped to this package, but since everything coming from IO is inherently going to involve an HTTP request, we don't really need to add that to the hierarchy.